### PR TITLE
[4.2] [Rector] Apply PHP 7.4 Requirement and syntax

### PR DESCRIPTION
**Pull request by samsonasik**
**Description**

@MGatner here the update to php 7.4:

- Upgrade to php 7.4 as minimum requirement.
- Update syntax with Rector

TODO: 

- [x] Fix PHPStan notice
- [x] Apply Workflow run

**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

**Comment by MGatner** [comment]
Looking good!

**Comment by samsonasik** [comment]
Temporary run rector on php 8.0 only due to error `Call to undefined method ReflectionProperty::getDefaultValue()`, fix is coming to rector https://github.com/rectorphp/rector-src/pull/1482

**Comment by samsonasik** [comment]
I temporary ignore phpstan notice on syntax `$var ??= $value` https://github.com/codeigniter4/CodeIgniter4/pull/5462/commits/71037bae81831e4765d80e9ae315474461afeba4

File: system/HTTP/URI.php [hunk 1]
```diff
@@ -1025,11 +1023,7 @@ protected function parseStr(string $query): array
         $return = [];
         $query  = explode('&', $query);
 
-        $params = array_map(static function (string $chunk) {
-            return preg_replace_callback('/^(?<key>[^&=]+?)(?:\[[^&=]*\])?=(?<value>[^&=]+)/', static function (array $match) {
-                return str_replace($match['key'], bin2hex($match['key']), $match[0]);
-            }, urldecode($chunk));
-        }, $query);
+        $params = array_map(static fn (string $chunk) => preg_replace_callback('/^(?<key>[^&=]+?)(?:\[[^&=]*\])?=(?<value>[^&=]+)/', static fn (array $match) => str_replace($match['key'], bin2hex($match['key']), $match[0]), urldecode($chunk)), $query);
```
Comment (paulbalandan) [inline system/HTTP/URI.php:1026]: Can you break this into many lines?

Comment (samsonasik) [reply system/HTTP/URI.php]: updated https://github.com/codeigniter4/CodeIgniter4/pull/5462/commits/dd95ae72be1b6f9a7019cd919fd94446d19a0757

**Review by paulbalandan — APPROVED** [review]

**Comment by kenjis** [comment]
Looks good!

**Review by MGatner — APPROVED** [review]

**Comment by samsonasik** [comment]
I am merging it ;). Thanks @paulbalandan @kenjis @MGatner for the review.

File: rector.php [hunk 2]
```diff
@@ -154,4 +154,8 @@
     $services->set(FuncGetArgsToVariadicParamRector::class);
     $services->set(MakeInheritedMethodVisibilitySameAsParentRector::class);
     $services->set(SimplifyEmptyArrayCheckRector::class);
+    $services->set(TypedPropertyRector::class)
+        ->configure([
+            TypedPropertyRector::PRIVATE_PROPERTY_ONLY => true,
```
Comment (samsonasik) [inline rector.php:159]: private property only option will be removed in next rector release https://github.com/rectorphp/rector-src/pull/1496, to have default change:

- change private property
- change protected property for final class.
- don't update public property.

We can retry re-run when on next rector release.

**Pull request merged**